doc/mgmt/SONiC_DNS_Support.md

```diff
@@ -0,0 +1,407 @@
+# Feature Name
+Domain Name System (DNS) support
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 05/05 |   Venkatesan Mahalingam         | Initial version                   |
+
+
+# About this Manual
+
+This document introduces the support of DNS in SONiC.
+
+# Scope
+
+This document covers the following,
+1) Click commands to configure DNS server & Source IP
+2) Click command to show DNS hosts
+3) New tables in config DB to handle DNS configs
+4) DNS table configuration and show based on OpenConfig YANG model
+5) KLISH commands to configure DNS server & Source IP
+6) KLISH commands to show DNS hosts
+7) Backend support to add DNS configs in "/etc/resolv.conf" file & iptable rules to change the source-IP of the DNS query packets
+8) Unit Testcases
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+|          DNS             |  Domain Name System             |
+
+
+
+# 1 Feature Overview
+
+DNS stands for Domain Name System. It is used to translate human readable domain names into machine readable IP addresses.
+
+With this feature, users will be able to configure DNS servers and source IP using SONiC Click commands and north bound interface (KLISH/REST/gNMI) provided by management framework module using OpenConfig models.
+
+Also, backend support to handle new config DB table events to populate nameservers in "/etc/resolv.conf" and add iptable rules to change the source IP of the DNS query that is being sent to the DNS server.

```

Comment:
We'll explore dnsmasq package for setting the source IP.

Reply:
dnsmasq does not work for loopback interface and hence adding this package to SONiC is required at this point.